server/map_rpc_server.go

```diff
@@ -306,6 +291,35 @@ func (t *TrillianMapServer) SetLeaves(ctx context.Context, req *trillian.SetMapL
 	return &trillian.SetMapLeavesResponse{MapRoot: newRoot}, nil
 }
 
+// txRunner satisfies the merkle.TXRunner interface.
+type txRunner struct {
+	tree       *trillian.Tree
+	tx         storage.MapTreeTX
+	txRolledUp uint64
+	mapStorage storage.MapStorage
+	opts       TrillianMapServerOptions
+}
+
+func (t *TrillianMapServer) newTXRunner(tree *trillian.Tree, tx storage.MapTreeTX) *txRunner {
+	return &txRunner{tree: tree, mapStorage: t.registry.MapStorage, opts: t.opts}
+}
+
+// RunTX runs a transaction given a transaction runner f.
+func (r *txRunner) RunTX(ctx context.Context, f func(context.Context, storage.MapTreeTX) error) error {
+	if r.opts.UseSingleTransaction {
+		glog.V(1).Infof("Using enclosing tx for subtree operation %d", atomic.LoadUint64(&r.txRolledUp))
+		atomic.AddUint64(&r.txRolledUp, 1)
+		return f(ctx, r.tx)
+	}
+	return r.mapStorage.ReadWriteTransaction(ctx, r.tree, f)
+}
+
+func (r *txRunner) Close() {
```

Comment:
You can get rid of this function and just ding the txRolledUp counter - it's not used anywhere.
(It's got potential for confusion currently since Close() doesn't actually close the transaction.)